File: app/src/main/java/ua/com/foxminded/locationtrackera/model/locations/dao/RoomLocationsDao.java
package ua.com.foxminded.locationtrackera.model.locations.dao;

import android.app.Application;

import java.util.List;

import ua.com.foxminded.locationtrackera.model.locations.UserLocation;

public class RoomLocationsDao implements LocationsDao {

    private final UserLocationDao userLocationDao;

    public RoomLocationsDao(Application application) {
        final LocationDatabase database = LocationDatabase.getInstance(application);
        userLocationDao = database.userLocationDao();
    }

    @Override
    public void saveLocation(UserLocation userLocation) {
        userLocationDao.saveLocation(userLocation);
    }

    @Override
    public List<UserLocation> getAllLocations() {
        return userLocationDao.getAllLocations();
    }

    @Override
    public void deleteAllLocation() {
        userLocationDao.deleteAllLocation();
    }
}

File: models_impl/src/main/java/ua/com/foxminded/locationtrackera/models_impl/locations/dao/RoomLocationsDao.kt
package ua.com.foxminded.locationtrackera.models_impl.locations.dao

import android.app.Application
import ua.com.foxminded.locationtrackera.models.locations.UserLocation

class RoomLocationsDao(application: Application) :
    ua.com.foxminded.locationtrackera.models.locations.dao.LocationsDao {

    private val userLocationDao: UserLocationDao

    init {
        val database = LocationDatabase.getInstance(application)
        userLocationDao = database.userLocationDao()
    }

    override fun saveLocation(userLocation: UserLocation) =
        userLocationDao.saveLocation(userLocation)

    override fun getAllLocations(): List<UserLocation> = userLocationDao.getAllLocations()

    override fun deleteAllLocation() = userLocationDao.deleteAllLocation()
}